pkg/pod/pod.go

```diff
@@ -26,36 +25,58 @@ type Client struct {
 type ClientInt interface {
 	GetPods() (pods []*v1.Pod, err error)
 	Start(exit <-chan struct{})
+	ListPods() (pods []*v1.Pod, err error)
 }
 
 // NewPodClient returns new pod client
 func NewPodClient(i informers.SharedInformerFactory, eventCh chan aadpodid.EventType) (c ClientInt) {
 	podInformer := i.Core().V1().Pods()
-	addPodHandler(podInformer, eventCh)
+	addPodHandler(podInformer, eventCh, nil)
+
+	return &Client{
+		PodWatcher: podInformer,
+	}
+}
+
+func NewPodClientWithPodInfoCh(i informers.SharedInformerFactory, eventCh chan aadpodid.EventType, podInfoCh chan *v1.Pod) (c ClientInt) {
+	podInformer := i.Core().V1().Pods()
+	addPodHandler(podInformer, eventCh, podInfoCh)
 
 	return &Client{
 		PodWatcher: podInformer,
 	}
 }
 
-func addPodHandler(i informersv1.PodInformer, eventCh chan aadpodid.EventType) {
+func addPodHandler(i informersv1.PodInformer, eventCh chan aadpodid.EventType, podInfoCh chan *v1.Pod) {
 	i.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				klog.V(6).Infof("Pod Created")
 				eventCh <- aadpodid.PodCreated
-
 			},
 			DeleteFunc: func(obj interface{}) {
 				klog.V(6).Infof("Pod Deleted")
 				eventCh <- aadpodid.PodDeleted
-
 			},
 			UpdateFunc: func(OldObj, newObj interface{}) {
-				// We are only interested in updates to pod if the node changes.
+				oldPod := OldObj.(*v1.Pod)
+				newPod := newObj.(*v1.Pod)
+
+				// This is to handle windows nmi by observing ip change
+				if oldPod.Status.PodIP != newPod.Status.PodIP {
+					klog.V(6).Infof("Pod IP Updated")
+					klog.Infof("Old Pod IP: %s, Current Pod IP: %s", oldPod.Status.PodIP, newPod.Status.PodIP)
```

Comment:
Why have two logging lines side by side when we can have one. 
Also the top line has verbosity level V=6, line at the bottom has no verbosity level and produces log output always.
I suggest merging the lines and making logging verbosity consistent.

As for TODO: I suggest sweeping through the code we added/modified and making logging more consistent. If information we are logging is not important we can log it at higher verbosity level so we have cleaner logs, but we can enable more verbose logs if we need by just restarting container with different verbosity level setting
BTW: if you do not parse `klog` flags then logging verbosity will always be default.

This is non-blocking comment. Can be fixed in later iteration

Reply:
This is for debug purpose. And I forgot to remove it. I will fix in next iteration.